ops_test.cc

```diff
@@ -356,6 +356,155 @@ void TestAllCreateDistribution() {
   TestCreateDistribution<5000>();
 }
 
+template <size_t kOuter, size_t kInner>
+CompressedArray<float, kOuter * kInner> GenerateMat(size_t offset) {
+  hwy::ThreadPool pool(1);
+  gcpp::CompressWorkingSet ws;
+  CompressedArray<float, kOuter * kInner> mat;
+  std::array<float, kOuter * kInner> content;
+  const float scale = 1.0f / kInner;
+  for (size_t i = 0; i < kOuter; i++) {
+    for (size_t j = 0; j < kInner; j++) {
+      content[i * kInner + j] = static_cast<float>((i + j + offset) * scale);
+    }
+  }
+  Compress(content, ws, mat, pool);
+  return mat;
+}
+
+template <size_t length>
+hwy::AlignedFreeUniquePtr<float[]> GenerateVec(size_t offset) {
+  hwy::AlignedFreeUniquePtr<float[]> vec = hwy::AllocateAligned<float>(length);
+  for (size_t idx = 0; idx < length; idx++) {
+    vec[idx] = static_cast<float>(idx + offset);
+  }
+  return vec;
+}
+
+template <size_t kOuter, size_t kInner>
+hwy::AlignedFreeUniquePtr<float[]> SimpleMatVecAdd(
+    const CompressedArray<float, kOuter * kInner>& mat,
+    const hwy::AlignedFreeUniquePtr<float[]>& vec,
+    const hwy::AlignedFreeUniquePtr<float[]>& add) {
+  hwy::AlignedFreeUniquePtr<float[]> uncompressed_mat =
+      hwy::AllocateAligned<float>(kOuter * kInner);
+  Decompress(mat, 0, uncompressed_mat.get(), kOuter * kInner);
+  hwy::AlignedFreeUniquePtr<float[]> out = hwy::AllocateAligned<float>(kOuter);
+  for (size_t idx_row = 0; idx_row < kOuter; idx_row++) {
+    out[idx_row] = add[idx_row];
+    for (size_t idx_col = 0; idx_col < kInner; idx_col++) {
+      out[idx_row] +=
+          uncompressed_mat[kInner * idx_row + idx_col] * vec[idx_col];
+    }
+  }
+  return out;
+}
+
+template <size_t length>
+void AssertClose(const hwy::AlignedFreeUniquePtr<float[]>& a,
+                 const hwy::AlignedFreeUniquePtr<float[]>& b) {
+  for (size_t idx = 0; idx < length; idx++) {
+    const float rel_abs_delta =
+        std::abs(a[idx] - b[idx]) / std::max(a[idx], b[idx]);
```

Comment:
Also abs of the max? What if both are negative?

Reply:
Thanks for catching that!